src/main/kotlin/net/skripthub/docstool/modals/SyntaxData.kt

```diff
@@ -1,81 +1,53 @@
+    ) {
+
     fun toMap(): Map<String, Any> {
         val map = LinkedHashMap<String, Any>()
-        addProperty(map, "ID", id!!)
-        addProperty(map, "Name", name!!)
-        if(description != null){
-            addArray(map, "Description", description!!)
-        }
-        if (examples != null){
-            addArray(map, "Examples", examples!!)
-        }
-        if(usage != null){
-            addProperty(map, "Usage", *usage!!)
-        }
-        if(since != null){
-            addArray(map, "Since", since!!)
-        }
-        if(returnType != null){
-            addProperty(map, "Return type", returnType!!)
-        }
-        if(changers != null){
-            addArray(map, "Changers", changers!!)
-        }
-        addArray(map, "Patterns", patterns!!)
-        if(eventValues != null){
-            addArray(map, "Event values", eventValues!!)
-        }
-        if (cancellable != null) {
-            map["Cancellable"] = cancellable!!
-        }
-        if (requiredPlugins != null){
-            addArray(map, "Required Plugins", requiredPlugins!!)
-        }
-        if (entries != null){
-            addArray(map, "Entries", entries!!)
-        }
+
+        addProperty(map, "id", id!!)
+        addProperty(map, "name", name!!)
+        addArray(map, "description", description)
+        addArray(map, "examples", examples)
+        usage?.let { addProperty(map, "usage", *it) }
+        addArray(map, "since", since)
+        addProperty(map, "return type", returnType)
+        addArray(map, "changers", changers)
+        addArray(map, "patterns", patterns)
+        addArray(map, "event values", eventValues)
+        cancellable?.let { map["cancellable"] = it }
+        addArray(map, "required plugins", requiredPlugins)
+        addEntryNodes(map, entries)
+        addArray(map, "keywords", keywords)
         return map
     }
 
-    private fun addArray(map: LinkedHashMap<String, Any>, property: String, array: Array<DocumentationEntryNode>) {
-        if (array.isEmpty())
-            return
-        map[property] = array
+    private fun addProperty(map: MutableMap<String, Any>, property: String, vararg values: String?) {
+        map[property] = if (values.isNotEmpty() && values.any { it.isNullOrBlank().not() }) StringUtils.join(values.filterNotNull(), "\n") else return
     }
 
-    private fun addProperty(map: MutableMap<String, Any>, property: String, vararg value: String) {
-        if (value.isEmpty())
-            return
-        val sb = StringBuilder()
-        for (str in value) {
-            if (str.isNotEmpty()) {
-                if (sb.isNotEmpty())
-                    sb.append("\n")
-                sb.append(str)
-            }
-        }
-        if (sb.isNotEmpty())
-            map[property] = sb.toString()
+    private fun addEntryNodes(map: MutableMap<String, Any>, entries: Array<DocumentationEntryNode>?) {
+        map["entries"] = if (entries.isNullOrEmpty().not()) entries!! else return
```

Comment:
If we can, let try to avoid using not-null assertion operator like `entries!!` here. You should be able to avoid it by using patterns like below:
```suggestion
        map["entries"] = if (!entries.isNullOrEmpty()) entries else return
```